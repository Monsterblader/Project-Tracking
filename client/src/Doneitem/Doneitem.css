.done-item .message {
  text-decoration: line-through;
}
