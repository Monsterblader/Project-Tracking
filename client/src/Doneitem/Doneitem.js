import React from 'react';
import Checkmark from '../assets/Checkmark.js';
import Xmark from '../assets/Xmark.js';
import './Doneitem.css';

function Doneitem(props) {
  return (
    <div className="done-item">
      <div
          className="svg-container"
          onClick={ () => {
            console.log('clicked');
            props.uncompleteItem();
          } }>
        <svg
            height="30.0"
            transform="scale(-1,1) translate(0,5)"
            width="30.0">
          <Checkmark />
        </svg>
      </div>
      <div
          className="svg-container"
          onClick={ props.delete }>
        <Xmark />
      </div>
      <div className="message text-box">
        { props.data.item }
      </div>
    </div>
  );
}

export default Doneitem;
