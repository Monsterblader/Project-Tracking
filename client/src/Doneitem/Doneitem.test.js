import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import Renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import Doneitem from './Doneitem.js';

jest.mock('../assets/Checkmark.js',
  () =>
    () => <div id="mock-Checkmark">Checkmark</div>
)
jest.mock('../assets/Xmark.js',
  () =>
    () => <div id="mock-Xmark">Xmark</div>
)

describe('Doneitem', () => {
  const data = {
    id: 1234,
    item: "test value",
  };
  const del = jest.fn();
  const uncompleteItem = jest.fn();
  let container = null;

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  test('it renders "Doneitem"', () => {
    const component = Renderer.create(
      <Doneitem data={ data } />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    render(<Doneitem data={ data } />, container);
  });

  it('creates an item with "test value"', () => {
    render(<Doneitem data={ data } />, container);
    const el = container.getElementsByClassName('message text-box')[0];
    expect(el.textContent).toEqual(data.item);
  });

  describe('the <Checkmark />', () => {
    it('renders <Checkmark />', () => {
      render(<Doneitem data={ data } />, container);
      expect(document.getElementById('mock-Checkmark')).toBeTruthy();
    });

    it('calls uncompleteItem on click', () => {
      render(<Doneitem data={ data } uncompleteItem={ uncompleteItem }/>, container);
      const el = document.getElementById('mock-Checkmark');
      el.parentNode.parentNode.click();
      expect(uncompleteItem).toHaveBeenCalledTimes(1);
    });
  });

  describe('the <Xmark />', () => {
    it('renders <Xmark />', () => {
      render(<Doneitem data={ data } />, container);
      expect(document.getElementById('mock-Xmark')).toBeTruthy();
    });

    it('calls delete on click', () => {
      render(<Doneitem data={ data } delete={ del }/>, container);
      const el = document.getElementById('mock-Xmark');
      el.parentNode.click();
      expect(del).toHaveBeenCalledTimes(1);
    });
  });
});
