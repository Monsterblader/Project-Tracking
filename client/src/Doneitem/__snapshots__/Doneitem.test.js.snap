// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doneitem it renders "Doneitem" 1`] = `
<div
  className="done-item"
>
  <div
    className="svg-container"
  >
    <svg
      height="30.0"
      transform="scale(-1,1) translate(5,5)"
      width="30.0"
    >
      <div
        id="mock-Checkmark"
      >
        Checkmark
      </div>
    </svg>
  </div>
  <div
    className="svg-container"
  >
    <div
      id="mock-Xmark"
    >
      Xmark
    </div>
  </div>
  <div
    className="message text-box"
  >
    test value
  </div>
</div>
`;
